internal/pkg/etcd/etcd.go

```diff
@@ -103,6 +107,36 @@ func (c *Client) ValidateForUpgrade(ctx context.Context, config config.Provider,
 				return fmt.Errorf("etcd member %d is not started, all members must be running to perform an upgrade", member.ID)
 			}
 		}
+
+		// Member count must not be an even number, lest the uprgade cause a loss of quorum.
+		if len(resp.Members) % 2 == 0 {
+			return fmt.Errorf("etcd member count (%d) is even; upgrade of this node may cause loss of quorum", len(resp.Members))
+		}
+
+		// Quorum must be good before allowing an upgrade of an etcd member.
+		if err = c.ValidateQuorum(ctx); err != nil {
```

Comment:
should we check quorum via each endpoint? this is what etcd docs recommend on upgrade.

We have a way to send requests via apid from the node to other nodes easily, so we can query Services API for every master node and process the result to figure out etcd service health

Reply:
I would not have presume that to be necessary, since etcd is both secured against split brain and kv operations are secured against loss of quorum.  If that is recommended, though, we should certainly do so.